ckanext/saml2/plugin.py

```diff
@@ -663,21 +713,17 @@ def staff_login(self):
         return self.login()
 
     def edit(self, id=None, data=None, errors=None, error_summary=None):
-        c = p.toolkit.c
-        c.allow_user_changes = p.toolkit.asbool(
-            config.get('ckan.saml2.allow_user_changes', False))
-        c.is_allow_update = saml2_get_user_info(c.user).first().allow_update
+        saml2_check_for_user_update(id)
```

Comment:
You have problem with naming. I do not see any result of check(i.e. return value), so i suppose, that this function will raise error if check fails